tests/test_30_mdstore.py

```diff
@@ -324,6 +325,27 @@ def test_mdx_single_sign_on_service():
     assert sso_loc[0]["location"] == "http://xenosmilus.umdc.umu.se/simplesaml/saml2/idp/metadata.php"
 
 
+@responses.activate
+@mock.patch('saml2.mdstore.before')
+def test_mdx_metadata_freshness_period(mock_datetime):
+    """Ensure that metadata is refreshed only when they have expired."""
+    entity_id = "http://xenosmilus.umdc.umu.se/simplesaml/saml2/idp/metadata.php"
+
+    url = "http://mdx.example.com/entities/{}".format(
+        parse.quote_plus(MetaDataMDX.sha1_entity_transform(entity_id)))
+    responses.add(responses.GET, url, body=TEST_METADATA_STRING, status=200,
+                  content_type=SAML_METADATA_CONTENT_TYPE)
+
+    mock_datetime.return_value = True
+    mdx = MetaDataMDX("http://mdx.example.com",
+                      freshness_period="P0Y0M0DT0H2M0S")
+    mdx.single_sign_on_service(entity_id, BINDING_HTTP_REDIRECT)
+    mdx.single_sign_on_service(entity_id, BINDING_HTTP_REDIRECT)
+    mock_datetime.return_value = False
+    mdx.single_sign_on_service(entity_id, BINDING_HTTP_REDIRECT)
+    assert len(responses.calls) == 2
```

Comment:
Why not add an assert after each call to ensure the calls are made when they should be?